include/CLI/App.hpp

```diff
@@ -1231,8 +1283,10 @@ class App {
     /// This must be called after the options are in but before the rest of the program.
     void parse(int argc, const char *const *argv) {
         // If the name is not set, read from command line
-        if(name_.empty())
+        if((name_.empty()) || (has_automatic_name_)) {
+            has_automatic_name_ = true;
```

Comment:
This only gets called on the main command, it can't be a subcommand. Would it be possible to avoid the `has_automatic_name_`?

Reply:
The issue that is solving is if one of the nameless subcommands were called in an independent context with argc, argv.    In that case it would prohibit it from being used as a nameless subcommand in later contexts unless the automatically generated name can be knowingly and automatically cleared.

Reply:
Okay; I like to limit global state if possible, but I think this one's okay.